src/v2/Apps/Conversation/Components/ConversationMessages.tsx

```diff
@@ -72,80 +74,81 @@ export const ConversationMessages = ({
       item?.__typename !== "Message" && relevantEvents.includes(item.__typename)
     )
   }
-  return <>
-    {messagesAndEvents.map((messageGroup, groupIndex) => {
-      let today: boolean
-      let newFlagShown = false
-      if (messageGroup[0].createdAt) {
-        today = fromToday(messageGroup[0].createdAt)
-      }
+  return (
+    <>
+      {messagesAndEvents.map((messageGroup, groupIndex) => {
+        let today: boolean
+        let newFlagShown = false
+        if (messageGroup[0].createdAt) {
+          today = fromToday(messageGroup[0].createdAt)
+        }
 
-      return (
-        <Fragment
-          key={`group-${groupIndex}-${messageGroup[0]?.internalID}`}
-        >
-          {messageGroup[0].__typename === "Message" && (
-            <TimeSince
-              style={{ alignSelf: "center" }}
-              time={
-                messageGroup[0].createdAt ? messageGroup[0].createdAt : ""
+        return (
+          <Fragment key={`group-${groupIndex}-${messageGroup[0]?.internalID}`}>
+            {messageGroup[0].__typename === "Message" && (
+              <TimeSince
+                style={{ alignSelf: "center" }}
+                time={
+                  messageGroup[0].createdAt ? messageGroup[0].createdAt : ""
+                }
+                exact
+                mb={1}
+              />
+            )}
+
+            {[...messageGroup].reverse().map((message, messageIndex) => {
+              if (isRelevantEvent(message)) {
+                return (
+                  <OrderUpdateFragmentContainer
+                    key={`event-${messageIndex}`}
+                    event={message as any}
+                    setShowDetails={setShowDetails}
```

Comment:
This is the only line that changed in the return statement. Everything else is prettified.